src/beacon-api/app/services/device_service.py

```
@@ -984,25 +979,20 @@ async def sync_devices(db: Session, token: str) -> Dict[str, Any]:
+                paged_responses = await asyncio.gather(*fetch_tasks)
+                for resp in paged_responses:
+                    if resp.status_code == 200:
+                        all_devices.extend(resp.json().get("devices", []))
         except Exception as e:
             logger.exception(f"Unexpected error fetching devices for sync: {e}")
-            return {
-                "success": False,
-                "message": "Network or parsing error while fetching devices for sync.",
-                "status_code": 500,
-            }
+            return {"success": False, "message": f"Network or Parsing error: {str(e)}", "status_code": 500}
```

Ruff (0.15.9)

[warning] Use explicit conversion flag

Replace with conversion flag

(RUF010)